packages/validate/src/lib/validation.ts

```diff
@@ -477,15 +666,21 @@ export const unknownObject = new Validator<Record<string, unknown>>((value) => {
 	return value as Record<string, unknown>
 })
 
+type ExtractRequiredKeys<T extends object> = {
+	[K in keyof T]: undefined extends T[K] ? never : K
+}[keyof T]
+
+type MakeUndefinedOptional<T extends object> = Pick<T, ExtractRequiredKeys<T>> & Partial<T>
+
```

Comment:
why was this needed? it seems to have made the hover-completions and generated types from validators a lot clunkier

Reply:
Might be worth wrapping it or its use in an `Expand` and seeing if that helps at all?